Feature/profile account

andrey-dihtiarov opened the pull request:
Добавил все нужные компоненты для profile формы (DateInput, RadioGroup и AddressInput). Но не добавлял все поля в профайл форму пока, т.к. решили, что нужно быстрее браться за step wizzard компонент.

Commit:
profile-account: date-fns added to dependencies list

Commit:
profile-account: DateInput component added

Commit:
profile-account: react-datepicker styles added to global styles

Commit:
profile-account: Birth date input field added to ProfileForm

Commit:
profile-account: validation for birthDate input added

andrey-dihtiarov committed:
profile-account: RadioInput component added

andrey-dihtiarov committed:
profile-account: RadioGroup comonent added

andrey-dihtiarov committed:
profile-account: gender field added to ProfileForm

andrey-dihtiarov committed:
profile-account: gender field validation added

andrey-dihtiarov committed:
profile-account: places js added to dependencies list

andrey-dihtiarov committed:
profile-account: AddressInput component added

andrey-dihtiarov committed:
profile-account: global styles updated

andrey-dihtiarov committed:
profile-account: AddressField added to ProfileForm

andrey-dihtiarov committed:
profile-account: initial value for address field added

oleksandrbazil commented on src/components/RadioGroup/index.jsx:
```diff
@@ -0,0 +1,32 @@
+
+  return (
+    <InputContainer field={field} label="Gender">
+      <div className={styles.radioGroup} onChange={onValueChange}>
```
а как onChange handler обрабатывается? это по документации так?
мне кажется тут что-то не так

andrey-dihtiarov replied:
Исправил

oleksandrbazil commented on src/components/RadioGroup/index.jsx:
```diff
@@ -0,0 +1,32 @@
+  return (
+    <InputContainer field={field} label="Gender">
+      <div className={styles.radioGroup} onChange={onValueChange}>
+        {values &&
```
нет смысла проверять есть ли values что бы отмаппить его

andrey-dihtiarov replied:
Точно. Убрал это

oleksandrbazil commented on src/components/RadioGroup/index.jsx:
```diff
@@ -0,0 +1,32 @@
+
+import styles from './styles.module.scss';
+
+const RadioGroup = ({ values, name }) => {
```
значение по умолчанию?

andrey-dihtiarov replied:
Не понял о чем речь. Уточни плиз

oleksandrbazil commented on src/components/forms/ProfileForm/index.jsx:
```diff
@@ -1,3 +1,17 @@
+import RadioGroup from '../../RadioGroup';
+import AddressInput from '../../inputs/AddressInput';
+
+const GenderRadioGroupValues = ['Male', 'Female'];
```
есть ли реальный смысл тут выносить это в константу?

andrey-dihtiarov replied:
Убрал константу

oleksandrbazil commented on src/components/inputs/AddressInput/index.jsx:
```diff
@@ -0,0 +1,37 @@
+        className={`${styles.field} ${isError && styles.fieldError}`}
+        type="search"
+        id="address-input"
+        placeholder="Where are we going?"
```
похоже как будь-то забыли убрать placeholder после копи-паста примера

andrey-dihtiarov replied:
Убрал

oleksandrbazil commented on src/components/inputs/AddressInput/index.jsx:
```diff
@@ -0,0 +1,37 @@
+    <InputContainer field={field} label={label}>
+      <input
+        {...field}
+        className={`${styles.field} ${isError && styles.fieldError}`}
```
в данном случае лучше писать тернарным оператором что бы не было слова undefined в классах
Проверь плз и другие компоненты

andrey-dihtiarov replied:
Да, спасибо. Не обратил на это внимание. Поправил

oleksandrbazil commented on src/components/inputs/RadioInput/index.jsx:
```diff
@@ -0,0 +1,8 @@
+const RadioInput = ({ field: { name, value }, label, className, checked, ...rest }) => (
```
а нам точно нужно выносить в отдельный компонент?

andrey-dihtiarov replied:
Убрал этот компонент

oleksandrbazil commented on src/components/inputs/RadioInput/index.jsx:
```diff
@@ -0,0 +1,8 @@
+  </div>
+);
+
+export default RadioInput;
```
не описаны пропсы

andrey-dihtiarov replied:
Убрал этот компонент

oleksandrbazil commented on src/index.scss:
```diff
@@ -1,3 +1,5 @@
+@import '~react-datepicker/dist/react-datepicker.css';
```
почему решил импортировать стили тут?

andrey-dihtiarov replied:
Хотелось бы все стили, которые не модули, имрортировать куда-то в одно место. Если нужно, то поправлю

oleksandrbazil commented on src/pages/users/UserFormPage/index.jsx:
```diff
@@ -55,18 +66,16 @@ const validationSchema = Yup.object().shape({
     .test('fileFormat', 'Unsupported Format', (value) =>
       value ? SUPPORTED_FORMATS.has(value.type) : true,
     ),
+  birthDate: Yup.date().max(subYears(Date.now(), 18), 'User must be older 18').required('Required'),
```
а зачем добавил эту библиотеку? думаю в тех задании не говорилось о ней

```*T ```

andrey-dihtiarov replied:
Речь про date-fns? Библиотеки для работы со временем нельзя использовать?

oleksandrbazil commented on src/pages/users/UserFormPage/index.jsx:
```diff
@@ -55,18 +66,16 @@ const validationSchema = Yup.object().shape({
       value ? SUPPORTED_FORMATS.has(value.type) : true,
     ),
+  birthDate: Yup.date().max(subYears(Date.now(), 18), 'User must be older 18').required('Required'),
+  gender: Yup.string().oneOf(['Male', 'Female']).required('Required'),
```
я ранее писал зачем выносил в константы (не использовались), теперь вижу что нужно выносить и переиспользовать тут
кроме этого сделай так что бы константы нельзя было изменить в runtime

andrey-dihtiarov replied:
Сделал

Commit:
profile-account: RadioGroupInput
- unnecessary onChange handler removed
- unnecessary value check removed
- unnecessary RadioInput Component removed
- RadioGroup renamed to RadioGroupInput and moved to inputs folder

Commit:
profile-account: AddressInput
- removed unnecessary placeholder
- input border radius fixed

Commit:
profile-account: removed GenderRadioGroupValues constant

Commit:
profile-account: isError checks used with ternary operator now

Commit:
profile-account: RadioGroup renamed to RadioGroupInput

Commit:
profile-account: gender input constants added, global constants is immutable now

Commit:
profile-account: DateInput and AddressInput width fix

Commit:
profile-account: Added all ProfileForm fields and validation

Commit:
profile-account: additional css container added

Commit:
profile-account: changed 'avatar' to dynamic value

Commit:
profile-account: AddressInput
- sets selected value now
- PropTypes added

Commit:
profile-account: submit button removed

andrey-dihtiarov committed:
profile-account: fieldsContainer width fixed

andrey-dihtiarov committed:
profile-account: TODOs added
